tasks/ca-certificates.yml

```diff
@@ -29,3 +29,18 @@
     keystore_pass: "{{ ca_certificates.password }}"
     state: present
   with_items: "{{ ca_certificates.certificates }}"
+
+# Workaround for https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=894979
+# This only happens on Stretch with Java 11 via stretch-backports.
+# This bug is fixed in ca-certificates-java in Buster.
+- name: Check format of keystore (stretch-backports)
+  command: "file /etc/ssl/certs/java/cacerts"
+  changed_when: false
+  check_mode: false
+  register: cacert_type
+  when: ansible_distribution_release == 'stretch' and 'openjdk-11-jdk' in java_packages
+
+- name: Migrate keystore to JKS (stretch-backports)
+  command: "keytool -importkeystore -srckeystore /etc/ssl/certs/java/cacerts -destkeystore /etc/ssl/certs/java/cacerts -srcstoretype PKCS12 -deststoretype JKS -srcstorepass {{ ca_certificates.password }} -deststorepass {{ ca_certificates.password }}"
+  changed_when: true
+  when: ansible_distribution_release == 'stretch' and 'openjdk-11-jdk' in java_packages and "data" in cacert_type.stdout
```

Comment:
What about Buster ?

Reply:
As said in the comment, this bug doesn't affect buster, as a fix to `ca-certificates-java` has been shipped with it. I have tested it manually with Docker, and the cacerts file with the newer format is properly created and works with the sample App without having to migrate cacerts to the old format.

Buster support will be added if needed in another PR.